modin/pandas/base.py

```diff
@@ -2213,6 +2214,30 @@ def rank(
             )
         )
 
+    def _copy_index_metadata(
+        self, source_index, destination_index
+    ):  # noqa: PR01, RT01, D200
+        """
+        Copy metadata from `source_index` to `destination_index` inplace.
+        """
+        if hasattr(source_index, "name") and hasattr(destination_index, "name"):
+            destination_index.name = source_index.name
+        if hasattr(source_index, "names") and hasattr(destination_index, "names"):
+            destination_index.names = source_index.names
```

Comment:
don't we also have to check for `.tz`, `.freq`?

Reply:
We have an example with `freq` in Index, as you can see from test, pandas does not preserve them. Therefore, Modin also should not preserve these attributes.